build_tools/benchmarks/collect_compilation_statistics_test.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import tempfile
+import unittest
+import zipfile
+
+from common.benchmark_definition import ModuleComponentSizes
+from collect_compilation_statistics import BENCHMARK_FLAGFILE, CONST_COMPONENT_NAME, VM_COMPONENT_NAME, get_module_component_info, get_module_path, match_module_cmake_target, parse_compilation_time_from_ninja_log
+
+
+class CollectCompilationStatistics(unittest.TestCase):
+
+  def test_match_module_cmake_target(self):
+    target = match_module_cmake_target(
+        "iree/iree-build/benchmark_suites/TFLite/vmfb/test.vmfb")
+
+    self.assertEqual(target, "benchmark_suites/TFLite/vmfb/test.vmfb")
+
+  def test_match_module_cmake_target_not_match(self):
+    target = match_module_cmake_target("benchmark_suites/TFLite/vmfb/test.mlir")
+
+    self.assertIsNone(target)
+
+  def test_parse_compilation_time_from_ninja_log(self):
+    target1 = "benchmark_suites/TFLite/vmfb/deeplabv3.vmfb"
+    target2 = "benchmark_suites/TFLite/vmfb/mobilessd.vmfb"
+    ninja_log = ("# ninja log v5\n"
+                 f"0\t100\taaa\tbuild/{target1}\taaa\n"
+                 f"130\t200\tbbb\tbuild/{target2}\tbbb\n")
+
+    with tempfile.NamedTemporaryFile("w") as f:
+      f.write(ninja_log)
+      f.flush()
+      target_map = parse_compilation_time_from_ninja_log(f.name)
```

Comment:
Probably better to extract the parsing logic from the file lookup so you don't have to mess with the filesystem in tests. If the method took TextIO instead of a path, then you could use string-backed IO here. Same below. I'm probably going to try to fix https://github.com/google/iree/issues/5147 during the fixit this week, so interacting with the filesystem in a test may be about to get harder :-)